debezium-core/src/main/java/io/debezium/relational/TableIdParser.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.debezium.text.ParsingException;
+import io.debezium.text.TokenStream;
+import io.debezium.text.TokenStream.CharacterStream;
+import io.debezium.text.TokenStream.Tokenizer;
+import io.debezium.text.TokenStream.Tokens;
+
+/**
+ * Parses identifiers into the corresponding parts of a {@link TableId}.
+ *
+ * @author Gunnar Morling.
+ */
+class TableIdParser {
+
+    private static final char SEPARATOR = '.';
+
+    public static List<String> parse(String identifier, char quotingChar) {
+        TokenStream stream = new TokenStream(identifier, new TableIdTokenizer(identifier, quotingChar), true);
+        stream.start();
+
+        List<String> parts = new ArrayList<>();
+
+        while(stream.hasNext()) {
+            parts.add(stream.consume());
+        }
+
+        return parts;
+    }
+
+    private static class TableIdTokenizer implements Tokenizer {
```

Comment:
I hear you and agree, I've already filed [DBZ-328](https://issues.jboss.org/browse/DBZ-328) to extend `BasicTokenizer` accordingly.

At this point I'd prefer to move forward with the proposed implementation though, for the sake of getting this into tomorrow's release. `BasicTokenizer` definitely is useful, unfortunately I missed some of its details originally. One minor concern is the usage of string builders for merging the current segment and also the nested looping which is avoided by the state machine approach but I suppose it's neglectable in practice.

Maybe you could propose a PR for adapting `BasicTokenizer`, supporting all the quoting characters and which passes all the cases in `TableIdParserTest` (there are some tests for error cases, such as leading or trailing separators which will need some more work in `BasicTokenizer` to be flagged correctly, which is what eventually made me sticking to the original implementation of this PR to get this out)?

Thanks a lot for the thorough reviewing and commenting!